scripts/balancing.lua

```diff
@@ -130,7 +150,7 @@ DefineUnitType("unit-water-elemental",{OnDeath = function(elemental, x, y)
       table.remove(CasterToElementalMap, caster)
       table.remove(ElementalToCasterMap, elemental)
       TransformUnit(caster, "unit-conjurer")
-      SetUnitVariable(caster, "Mana", 0)
+     -- SetUnitVariable(caster, "Mana", 0)
```

Comment:
So you're saying this crashes the game? Hmmm. Can you try with `SetUnitVariable(caster, "Mana", 1)` maybe? The idea here was that you cannot immediately cast elemental/demon again. Because if the demon survives long enough, the mana is almost refilled when it's gone, and then it can almost cast again immediately

Reply:
yes, i just check again now.
when you summon elemental, conjurer mana is not raising above 0 when he is meditating, until the elemental is dead, then it start raising again. it works well without this line.

with this mana set to 1. warlock is starting with 1 mana instead 0 after deamon is killed
im not sure why you want to keep it? in case someone reduce deamon cost below 100?

anyway with 1 or without this line it works well

Reply:
Ok, if it works like it is, we can merge this :)

Reply:
hmm... i just have CTD after casting deamon, deamon was killed in combat, but then warlock stay in meditation. It happen sometimes, im not sure whats the rule.
I click ESC on warlock, and then game crash.

SetUnitVariable(caster, "Mana", 0) is not a problem here

the problem is when you want to do break summoning, but there is no more deamon, because he was killed. so you click esc... but there is no deamon to remove.

hmm... maybe there need to be another check step? if there is NO deamon let warlock stop meditating?